source/security/authentication.txt

```diff
@@ -0,0 +1,390 @@
+but change the value of the ``authMechanism`` option to ``'SCRAM-SHA-1'``.
+
+.. _php-mongodb-x509:
+
+MONGODB-X509
+------------
+
+If you enable TLS, during the TLS handshake, the {+php-library+} can present an X.509
+client certificate to MongoDB to prove its identity. The MONGODB-X509 authentication
+mechanism uses this certificate to authenticate the client.
+
+To authenticate with this mechanism, set the following connection options:
+
+- ``tls``: Set to ``true``.
+- ``tlsCertificateKeyFile``: The file path of the ``.pem`` file that contains your
+  client certificate and private key.
+- ``authMechanism``: Set to ``'MONGODB-X509'``.
+
+You can set these options in two ways: by passing an options array to the
+``MongoDB\Client`` constructor or through parameters in your connection URI.
+
+.. tabs::
+
+   .. tab:: MongoDB\\Client
+      :tabid: Client
+
+      .. literalinclude:: /includes/authentication.php
+         :language: php
+         :dedent:
+         :start-after: start-mongodb-X509-client
+         :end-before: end-mongodb-X509-client
+
+   .. tab:: Connection URI
+      :tabid: connectionstring
+
+      .. literalinclude:: /includes/authentication.php
+         :language: php
+         :dedent:
+         :start-after: start-mongodb-X509-uri
+         :end-before: end-mongodb-X509-uri
+
+.. _php-mongodb-aws:
+
+MONGODB-AWS
+-----------
+
+.. important::
+
+   The MONGODB-AWS authentication mechanism requires {+mdb-server+} v4.4 or later.
+
+The MONGODB-AWS authentication mechanism uses AWS IAM (Amazon Web Services Identity and
+Access Management) or AWS Lambda credentials to authenticate your application. To use this
+method to authenticate, you must specify ``'MONGODB-AWS'`` as the value of
+the ``authMechanism`` connection option.
+
+.. note::
+
+   The {+php-library+} uses the C driver's implementation of the MONGODB-AWS
+   authentication mechanism. To learn more about using this authentication mechanism
+   with the C driver, see `Authentication via AWS IAM
+   <https://www.mongodb.com/docs/languages/c/c-driver/current/libmongoc/authentication/#authentication-via-aws-iam>`__
+   in the C documentation.
+
+When you use the MONGODB-AWS mechanism, the C driver tries to retrieve AWS
+credentials from the following sources, in the order listed:
+
+1. Options parameter passed to the ``MongoDB\Client`` constructor or parameters in the
+   connection URI
+#. Environment variables
+#. AWS EKS ``AssumeRoleWithWebIdentity`` request
+#. ECS container metadata
+#. EC2 instance metadata
+
+The following sections describe how to retrieve credentials from
+these sources and use them to authenticate your PHP application.
+
+.. _php-mongodb-aws-credentials:
+
+MongoDB\Client Credentials
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+First, the C driver checks whether you passed AWS credentials to the
+``MongoDB\Client`` constructor, either as an options parameter or as part of the
+connection URI. To pass your credentials to ``MongoDB\Client``,
```

Comment:
Consider:

> either as part of the connection string or `$uriOptions` array parameter

If "connection URI" is preferred, that's fine. I'd lead with talking about the first URI parameter, though, and then mention the options array.